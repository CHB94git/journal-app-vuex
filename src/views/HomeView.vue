<template>
  <div class="home justify-content-center">
    <img alt="Vue logo" src="../assets/logo.png">
    <button @click="goToDayBook" class="btn btn-primary">Ir a Journal</button>
  </div>
</template>


<script>
export default {
  methods: {
    goToDayBook () {
      this.$router.push({ name: 'no-entry' })
    }
  }
}
</script>

