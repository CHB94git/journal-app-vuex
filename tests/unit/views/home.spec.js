import { shallowMount } from '@vue/test-utils';
import HomeView from '@/views/HomeView'


describe('Pruebas componente HomeView', () => {
    it('Debe de renderizarse correctamente el componente', () => {
        const wrapper = shallowMount(HomeView)
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('hacer click en un botÃ³n debe de redireccionar a no-entry', () => {
        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(HomeView, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
    });
});