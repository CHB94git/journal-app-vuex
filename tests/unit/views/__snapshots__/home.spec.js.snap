// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente HomeView Debe de renderizarse correctamente el componente 1`] = `<div class="home justify-content-center"><img alt="Vue logo" src=""><button class="btn btn-primary">Ir a Journal</button></div>`;
